src/js/messaging/components/compose/MessageSend.jsx

```diff
@@ -8,13 +8,54 @@ class MessageSend extends React.Component {
   constructor(props) {
     super(props);
     this.handleAttachmentsChange = this.handleAttachmentsChange.bind(this);
+    this.validateNumAttachments = this.validateNumAttachments.bind(this);
+    this.validateFileSize = this.validateFileSize.bind(this);
+    this.validateTotalFileSize = this.validateTotalFileSize.bind(this);
+  }
+
+  validateNumAttachments(files, maxAttachments) {
+    return files.length > maxAttachments;
+  }
+
+  validateFileSize(files, max) {
+    return !!files.find((file) => { return file.size > max; });
+  }
+
+  validateTotalFileSize(files, max) {
+    // Get sizes for each file.
+    const sizes = files.map((f) => {
+      return f.size;
+    });
+
+    const total = sizes.reduce((a, b) => {
+      return a + b;
+    });
+
+    return total > max;
   }
 
   handleAttachmentsChange(domEvent) {
     const input = domEvent.target;
+    let hasError = null;
+
     if (window.File && window.FileList) {
-      const files = Array.from(input.files);
-      this.props.onAttachmentUpload(files);
+      if (input.files.length) {
+        const files = Array.from(input.files);
+
+        if (this.validateNumAttachments(files, this.props.maxFiles)) {
+          hasError = { type: 'tooMany' };
+        } else if (this.validateFileSize(files, this.props.maxFileSize) || this.validateTotalFileSize(files, this.props.maxTotalFileSize)) {
+          hasError = { type: 'tooLarge' };
```

Comment:
Would be good at some point to import literals like these as constants from config.